src/data_structures/arrays/ReverseString.java

```diff
@@ -15,8 +15,17 @@ public static String ReverseStringWithReelJava(String string) {
     return new StringBuilder(string).reverse().toString();
   }
 
+  public static String recursiveReversal (String str) {
+    if (str.length() == 0) {  //breakpoint for recursion
+       return "";
+    } else {
+       return (recursiveReversal(str.substring(1)) + str.charAt(0));
```

Comment:
Redundant outer brackets

Reply:
It is needed. Try some test input and check if this function is good.

Reply:
Java does not require brackets on returned values, not sure why you would think this is mandatory, the `+` operation will be calculated before the value is returned anyway.

Just for the sake of it I did run tests with those changes and got the same results.

Reply:
Oh Right.. Java does support indentation. But I prefer braces.. just to make it less confusing and avoid mistakes.. I'm old-school :) ..  
I am not understanding -> "+ operation will be calculated" part.. can you propose the code change. I will take a look.